python/adelphi/adelphi/gemini.py

```diff
@@ -26,7 +26,7 @@ def export_gemini_schema(keyspace_objs, metadata, options):
     # set replication factor
     set_replication_factor(keyspace_objs, options['rf'])
 
-    keyspace = keyspace_objs[0]
+    keyspace = list(keyspace_objs)[0]
```

Comment:
Python newbie here 👋 so you all tell me if you think this is the *right* way to fix this little problem @absurdfarce @grighetto

Reply:
Bah, looks like the view on dictionary keys behaves differently in Python3.  I don't know why I find this surprising... :(

The underlying issue here has to do with a change in the arg passed to export_gemini_schema(), and that same arg can be passed to any function with the responsibility for formatting output.  As a result I'd rather fix this the next level up (i.e. at the calling location in bin/adelphi) in order to provide uniform behaviour to the formatting functions.  I'm envisioning something like this:

[issue76_fix.txt](https://github.com/datastax/adelphi/files/5700596/issue76_fix.txt)

I can add this to the work I'm doing on the 67-naming-is-hard branch if you'd like.  Otherwise I'm happy to accept a pull request. ;)

Reply:
I'd definitely rather address it in the broader way that you mention, so I'll close this PR and you can include the fix.

Reply:
@absurdfarce I just wanted to point out that `list(...)` would work with lists, dict_keys, dict_values, whereas the patch with `next(...)` will break https://github.com/datastax/adelphi/pull/71/files#diff-f255dd10d942d71738355e9d80cd3e7cf64c68fd2b792492004875202eb928b0R209.

For a more general solution, I actually think we should probably revisit that `export_keyspaces` function to avoid building single-element lists and then picking just the first element in other places. I guess we can probably move the print logic into the for loop and process one keyspace at a time.